src/app/Fake.BuildServer.GitLab/GitLab.fs

```diff
@@ -186,11 +186,15 @@ module GitLab =
                 | TraceData.LogMessage(text, newLine) | TraceData.TraceMessage(text, newLine) ->
                     write false color newLine text
                 | TraceData.OpenTag (tag, descr) ->
+                    let unixTimestamp = System.DateTimeOffset.UtcNow.ToUnixTimeSeconds()
+                    let sectionHeader = sprintf "section_start:%d:%s_%s\r\e[0K%s" unixTimestamp tag.Type tag.Name
                     match descr with
-                    | Some d -> write false color true (sprintf "Starting %s '%s': %s" tag.Type tag.Name d)
-                    | _ -> write false color true (sprintf "Starting %s '%s'" tag.Type tag.Name)  
+                    | Some d -> write false color true (sectionHeader d)
+                    | None -> write false color true (sectionHeader "")
                 | TraceData.CloseTag (tag, time, state) ->
-                    write false color true (sprintf "Finished (%A) '%s' in %O" state tag.Name time)
+                    let unixTimestamp = System.DateTimeOffset.UtcNow.ToUnixTimeSeconds()
+                    let sectionFooter = sprintf "section_end:%d:%s_%s\r\e[0K" unixTimestamp tag.Type tag.Name
```

Comment:
I tried to use the `time` field here, but the problem with that is that the expected format of the closing tag doesn't want the duration, it just wants the unix ticks.  The TraceData types might be changed to include timestamp information (ie OpenTag gains a DateTimeOffset for the start, and CloseTag gains a DateTimeOffset and TimeSpan for the stop and duration respectively) and then this output could be fully-deterministic.  I'm not sure if you want to make that kind of a change or not.

Reply:
I don't mind it as-is. I'm not sure what you mean with "fully-deterministic". That output will never be deterministic as the time will always vary, correct?
So I don't see any gain from breaking anything here?
I guess you mean deterministic across different loggers?